api/v1alpha1/validation.go

```diff
@@ -116,3 +135,52 @@ func (r ResourceSync) Validate() []error {
 	allErrs = append(allErrs, validation.ValidateString(&r.Spec.Path, "spec.path", 0, 2048, nil, "")...)
 	return allErrs
 }
+
+func validateGitHttpConfig(config *GitHttpConfig) []error {
+	var errs []error
+	if config != nil {
+		if config.CaCrt != nil {
```

Comment:
Could we have a generic `ValidateBase64Field(field *string, maxLen int)` in util that in addition to testing for nil and base64 also tests for a max length constraint? That'd make this code more compact.